tests/bgp/test_bgp_chasssis.py

```diff
@@ -0,0 +1,424 @@
+            verify_routes_af(duts, allduts, ipversion, neigh_ip_addrs,
+                             neigh_intfs_by_dut, deleted=withdraw)
+            return (neigh_intfs_by_dut, neigh_ip_addrs)
+
+        def update_and_verify_routes(duts, allduts, num_nexthops_per_dut=None,
+                                     skip_v6=False, withdraw=False):
+            for ipversion in ipversions:
+                if skip_v6 and ipversion == IPV6:
+                    continue
+                update_and_verify_routes_af(duts, allduts, num_nexthops_per_dut,
+                                            ipversion, withdraw)
+
+        def check_intf_status(dut, dut_intf, exp_status):
+            status = dut.show_interface(command='status',
+                                        interfaces=[dut_intf])['ansible_facts']['int_status']
+            logging.info("status: %s", status)
+            return status[dut_intf]['oper_state'] == exp_status
+
+        def set_peerintf_state(dut, intf, intfstate):
+            cfg_facts = all_cfg_facts[dut.hostname][0]['ansible_facts']
+            intflist = []
+            if "portchannel" in intf.lower():
+                pc_cfg = cfg_facts['PORTCHANNEL_MEMBER']
+                pc_members = pc_cfg[intf]
+                intflist = pc_members.keys()
+            else:
+                intflist = [intf]
+
+            for lport in intflist:
+                fanout, fanport = fanout_switch_port_lookup(fanouthosts, dut.hostname, lport)
+                if intfstate == 'down':
+                    fanout.shutdown(fanport)
+                else:
+                    fanout.no_shutdown(fanport)
+
+                pytest_assert(wait_until(90, 1, check_intf_status, dut, intf, intfstate),
+                              "dut port {} didn't change state to {} as " \
+                              "expected".format(intf, intfstate))
+
+        def verify_bgp_inferior_path_propagation(dut, duthosts):
+            def verify_bgp_inferior_path_propagation_af(ipversion):
+                exabgp_ports = []
+                neigh_ip_addrs = []
+                neigh_intfs_by_dut = {}
+                num_paths = 2
+                for neigh_index in range(num_paths):
+                    get_neigh_info(dut, exabgp_ports, neigh_ip_addrs,
+                                   neigh_intfs_by_dut, ipversion, neigh_index)
+                # Adverties two path, one inferior to the other. Verify both paths
+                # are propagated and the bext path is chosen.
+                bestpath_index = 0
+                inferior_path_index = 1
+                inferior_paths_nexthops = [neigh_ip_addrs[inferior_path_index]]
+                update_routes(exabgp_ports[bestpath_index], routes[ipversion])
+                update_routes(exabgp_ports[inferior_path_index], routes[ipversion],
+                              asn=str(DUMMY_ASN1))
+                otherduts = [d for d in duthosts if d != dut and not d.is_supervisor_node()]
```

Comment:
You could use duthosts.frontend_nodes as otherduts, instead of the check of is_supervisor_node()